src/BitOasis/Coin/Mapping/CoinObjectHydrationListener.php

```diff
@@ -85,7 +86,9 @@ public function postLoad($entity, LifecycleEventArgs $args) {
 				if($coinClass->getFieldMapping($coinField)['type'] === CoinType::COIN) {
 					$coinClass->setFieldValue($entity, $coinField, Coin::fromInt($value, $currency));
 				} else {
-					$coinClass->setFieldValue($entity, $coinField, $this->cryptocurrencyAddressFactory->deserialize($value, $currency));
+					// TODO: Is this correct?
+					$network = $entity->cryptocurrencyNetwork ?? CryptocurrencyNetworkMapping::getDefaultNetworkForCurrency($currency);
```

Comment:
This might work, but.

What if the field is not named "cryptocurrencyNetwork"?
Even for "Coin" field you can have annotation with param cryptocurrency="myCustomCryptocurrencyFieldName"
For example:
```
	/**
	 * @ORM\Column(type="coin", options={"cryptocurrency": "feeCryptocurrency"})
	 * @var Coin
	 */
	protected $feeAmount;
```

this is handled in \BitOasis\Coin\Mapping\CoinObjectHydrationListener::buildCoinFields (lookup for the field name, default fallback). And this should be similar for networks. Please note that only address needs network, not coin.